xdsl/dialects/arm_neon.py

```diff
@@ -252,20 +252,25 @@ class DSSFmlaVecScalarOp(ARMInstruction):
     """
 
     name = "arm_neon.dss.fmla"
-    d = result_def(NEONRegisterType)
+    res = result_def(NEONRegisterType)
+    d = operand_def(NEONRegisterType)
```

Comment:
These have to be the same type. I would add this as a constraint and to the documentation, as well as a test verifying this

Reply:
See `RdRsAccumulatingFloatOperation`

Reply:
should this be a unit test or filecheck?

Reply:
ah right, I missed this. I think filecheck might be a bit easier